libp2p/pubsub/pubsub_notifee.py

```diff
@@ -56,7 +69,17 @@ def __init__(
         :param network: network the connection was opened on
         :param conn: connection that was opened
         """
-        await self.dead_peers_queue.put(conn.muxed_conn.peer_id)
+        async with self.dead_peers_queue_lock:
+            try:
+                await self.dead_peers_queue.send(conn.muxed_conn.peer_id)
+            except (
+                trio.BrokenResourceError,
+                trio.ClosedResourceError,
+                trio.BusyResourceError,
+            ):
+                # Raised when the receive channel is closed.
+                # TODO: Do something with loggers?
+                ...
```

Comment:
probably want to get some form of logging in here before merging so this doesn't silently fail somehow....

Reply:
Update: here I only catch `trio.BrokenResourceError`, which occurs when `Pubsub` exits and the `receive_channel` is closed.